### PR TITLE
[WIP RFC] feat: enable eslint-plugin-tsdoc

### PR DESCRIPTION
**Pull request by raymondfeng**
See https://www.npmjs.com/package/eslint-plugin-tsdoc

Proposed by @bajtos 

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by bajtos** [comment]
@raymondfeng Is this superseding https://github.com/strongloop/loopback-next/pull/5375 by @frbuceta ?

File: acceptance/extension-logging-fluentd/src/__tests__/acceptance/fluent.acceptance.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ describe('LoggingComponent', function () {
 
   /**
    * Read fluentd log files to match against the given regular exp
-   * @param regex A regular expression for assertion
+   * @param regex - A regular expression for assertion
```
Comment (bajtos) [inline acceptance/extension-logging-fluentd/src/__tests__/acceptance/fluent.acceptance.ts:53]: I'd prefer to use the short syntax ("optional hyphens" from https://github.com/microsoft/tsdoc/pull/206) and configure the linter to enforce this style everywhere.

```suggestion
   * @param regex A regular expression for assertion
```

However, it seems that `tsdoc` is preferring the hyphen-based flavor you have in your pull request, so I guess it's best to follow the idiomatic style and don't invent our own conventions 🤷

File: packages/eslint-config/eslintrc.js [hunk 2]
```diff
@@ -154,6 +159,7 @@ module.exports = {
     '@typescript-eslint/prefer-nullish-coalescing': 'error',
     '@typescript-eslint/no-extra-non-null-assertion': 'error',
     '@typescript-eslint/return-await': 'error',
+    'tsdoc/syntax': 'warn',
```
Comment (bajtos) [inline packages/eslint-config/eslintrc.js:162]: Please use `error` to fail the build on tsdoc syntax errors. Most people ignore warnings (one of the reasons are that CI will pass and won't tell you on GitHub about the warnings), so over time we will end up with `npm lint` printing a lot of warnings.

File: examples/passport-login/src/authentication-strategies/facebook.ts [hunk 3]
```diff
@@ -3,16 +3,15 @@
...
-import {mapProfile} from './types';
+import { asAuthStrategy, AuthenticationStrategy } from '@loopback/authentication';
+import { StrategyAdapter } from '@loopback/authentication-passport';
+import { bind, inject } from '@loopback/context';
+import { extensionFor } from '@loopback/core';
+import { RedirectRoute, Request } from '@loopback/rest';
+import { UserProfile } from '@loopback/security';
+import { Strategy } from 'passport-facebook';
+import { User } from '../models';
+import { mapProfile, PassportAuthenticationBindings } from './types';
```
Comment (bajtos) [inline examples/passport-login/src/authentication-strategies/facebook.ts:14]: This is not following our formatting style, did you forgot to run prettier after re-organizing the imports?

File: examples/access-control-migration/src/__tests__/acceptance/project.acceptance.ts [hunk 4]
```diff
@@ -78,8 +78,8 @@ describe('AccessControlApplication - permissions', () => {
   /**
    * Test a role's permission when visit the 5 endpoints in the
    * project controller
-   * @param role
-   * @param permissions
+   * @param role - role
```
Comment (bajtos) [inline examples/access-control-migration/src/__tests__/acceptance/project.acceptance.ts:81]: I find these comments as not very useful. Describing a `role` parameter as "role" is not adding any new information for readers.

Why do we need this change? Is tsdoc linter complaining about params with no description? If that's true, then can we simply remove the `@param role` line entirely?

Maybe it's a good thing that the linter is complaining about missing description and we should use this as an opportunity to improve our API docs?

File: acceptance/extension-logging-fluentd/src/__tests__/acceptance/fluent.acceptance.ts (hunk 1 above)
Comment (raymondfeng) [reply acceptance/extension-logging-fluentd/src/__tests__/acceptance/fluent.acceptance.ts:53]: Yeah, the plugin warns about the missing `-`.

File: packages/eslint-config/eslintrc.js (hunk 2 above)
Comment (raymondfeng) [reply packages/eslint-config/eslintrc.js:162]: I'm testing with `warn` for now and will switch to `error` once we fix all issues.

File: examples/passport-login/src/authentication-strategies/facebook.ts (hunk 3 above)
Comment (raymondfeng) [reply examples/passport-login/src/authentication-strategies/facebook.ts:14]: Probably. Let me reformat.

File: examples/access-control-migration/src/__tests__/acceptance/project.acceptance.ts (hunk 4 above)
Comment (raymondfeng) [reply examples/access-control-migration/src/__tests__/acceptance/project.acceptance.ts:81]: This was a quick dirty fix. I was trying to see what else is left to be fixed.

**Pull request closed without merging**